javascript/code-challenges/challenges-14.test.js

```diff
@@ -0,0 +1,303 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function named screenForNames that takes in an array of strings and uses Regex to create a new array of only those strings that match the following rules:
+
+* the name must begin with Mr., Mrs., Ms., Dr. followed by a space
+* the name must contain only letter characters (white spaces are ok)
+
+------------------------------------------------------------------------------------------------ */
+
+const screenForNames = (arr) => {
+  const filter = /^[a-zA-z\.\s]*$/;
+  let retval = [];
+  for (let i in arr) {
+    const str = arr[i];
+    if (filter.test(str)) {
+      if (/^Mr. /.test(str)) {
+        if (str.charAt(4) !== ' ')
+          retval.push(str);
+      } else if (/^Mrs. /.test(str)) {
+        if (str.charAt(5) !== ' ')
+          retval.push(str);
+      } else if (/^Ms. /.test(str)) {
+        if (str.charAt(4) !== ' ')
+          retval.push(str);
+      } else if (/^Dr. /.test(str)) {
+        if (str.charAt(4) !== ' ')
+          retval.push(str);
+      }
+    }
+  }
+  return retval;
```

Comment:
Good work on finding a solution!

 My only feedback here is that we might consider finding expression that would cover all rules in one go round (to help simplify this a bit). We could then use `filter()` on the input str and `test()` to return an array that matches the given criteria.

Just some food for thought here 👍